### ISSUE TITLE
Windows fatal exception: access violation

### ISSUE THREAD

**Issue by ccampbell9** [qgis/QGIS#53091]
### What is the bug or the crash?

Multiple users in my organization are experiencing crashes with the below error output. All users are on the same model of machine; all are on the same version of QGIS; all instances of QGIS are newly installed with no non-stock plugins installed; all users are on windows 10.

```
Python Stack Trace
Windows fatal exception: access violation

Thread 0x00003908 (most recent call first):

Stack Trace
memcpy memcpy.asm:254
QgsRasterIdentifyResult::params :
QgsRasterIdentifyResult::operator= :
QgsSingleBandGrayRenderer::block :
QgsBrightnessContrastFilter::block :
QgsHueSaturationFilter::block :
QgsRasterResampleFilter::block :
QgsRasterProjector::block :
QgsRasterIterator::readNextRasterPartInternal :
QgsRasterDrawer::draw :
QgsRasterDrawer::draw :
QgsRasterLayerRenderer::render :
QgsMapRendererParallelJob::renderLayerStatic :
QgsProfilePlotRenderer::replaceSourceInternal :
QgsMapRendererParallelJob::renderingFinished :
QgsMapRendererParallelJob::takeLabelingResults :
QtConcurrent::ThreadEngineBase::run :
QThreadPoolPrivate::reset :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :

QGIS Info
QGIS Version: 3.30.2-'s-Hertogenbosch
QGIS code revision: 0992b533
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.6.4
Running against GDAL: 3.6.4

System Info
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.19045
```

### Steps to reproduce the issue

Users are not reporting any consistent pattern with the crashes, but they are increasing in frequency. 

Currently the users are digitizing/editing features in a PostGIS table and loading/rendering a raster vrt from an S3 bucket using the AWS S3 protocol. Our Mac and Linux users are not experiencing this problem.

Affected user system information:
Hardware: Dell Latitude 7420
CPU: Intel i7-1185G7
RAM: 16 GB
OS: Windows 10 (no updates available)   

### Versions

```
QGIS Info
QGIS Version: 3.30.2-'s-Hertogenbosch
QGIS code revision: 0992b533
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.6.4
Running against GDAL: 3.6.4
```

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

Possibly related to https://github.com/qgis/QGIS/issues/47888 but that is speculation.

**Comment by ccampbell9** [comment]
Possibly related to #47888 but that is speculation.

**Comment by elpaso** [comment]
Hard to track this down: any chance you could provide a sample project and data and a step by step procedure to reproduce the issue?

**Comment by jonnyforestGIS** [comment]
@elpaso I add the same problem but with other behaviour. In my case was in a vector temporary layer. 

The steps that I made:
In a vector temporary file without geometry went to the vector proprieties and in the tab of the fields, I toggle on the pencil and delete a field. 
Then I toggle off the field edition and toggle on again and delete 2 more fields and a crash occurs when I try to toggle off again with the same information error of this ticket.

I did a quick search and found this explanation in the Python forum:
https://discuss.python.org/t/windows-fatal-exception-access-violation/19591/3

Based on this threat someone says:

_The virtual address space of a process is allocated and managed in pages (typically 4 KiB blocks). A page can allow a combination of read, write, and execute access, or it may allow no access at all or no access to user-mode code (i.e. a page in kernel space). If an attempt to access an address in memory fails because the address isn’t allocated or because the type of access isn’t allowed for the page in which it resides, the Windows kernel raises an access violation exception on the faulting thread. Tracking down the cause usually requires testing a debug build, attaching a native debugger to the process, and running with special options such as full page-heap protection. Note that the immediate problem of the access violation may only be a symptom of a bug that leads to a bad pointer value, which could be indirectly from stack or heap corruption_.


Talking about Microsoft Windows Systems itself, another thing that can be related is the OS security permissions  (UAC or DEP):
https://www.makeuseof.com/how-to-fix-exception-access-violation/ (@ccampbell9 See problems fix 2 and 3 in the section How to Fix Exception Access Violation Error on Windows 10 and 11). 

Probably is not the solution but some hints to try to knock out this kind of issue.

Cheers
